src/index/mod.rs

```diff
@@ -173,6 +173,23 @@ impl Index {
     pub fn get_data_block(&self, index: usize) -> Option<&DataBlock> {
         self.data_blocks.get(index)
     }
+
+    // Get the data type for the given channel.
+    ///
+    /// Returns None if the channel does not exist.
+    pub fn channel_type(&self, path: &ChannelPath) -> Option<&DataType> {
+        self.objects
+            .get(path.path())
+            .and_then(|object| {
+                object
+                    .latest_data_format
+                    .as_ref()
+                    .and_then(|format| match format {
+                        DataFormat::RawData(meta) => Some(&meta.data_type),
+                    })
+            })
+    }
+
```

Comment:
@JamesMc86 new function allows getting a channel_type from an `Index` so that it can be accessed from higher level APIs